projects/plugins/jetpack/class.jetpack.php

```diff
@@ -5262,10 +5265,10 @@ public function post_login_form_to_signed_url( $url, $path, $scheme ) {
 		if ( 'wp-login.php' !== $path || ( 'login_post' !== $scheme && 'login' !== $scheme ) ) {
 			return $url;
 		}
-
-		$parsed_url = wp_parse_url( $url );
-		$url        = strtok( $url, '?' );
-		$url        = "$url?{$_SERVER['QUERY_STRING']}";
+		$query_string = isset( $_SERVER['QUERY_STRING'] ) ? sanitize_text_field( wp_unslash( $_SERVER['QUERY_STRING'] ) ) : '';
```

Comment:
`sanitize_text_field()` will probably break things here, as it strips out any percent-encoded characters.

(I really kind of hate WordPress's "sanitize" functions for that sort of reason)

Reply:
2dc7e0d8c1a9999981b1cd3b2d05279a6b94fdbb